sunpy/timeseries/timeseriesbase.py

```diff
@@ -623,20 +643,8 @@ def to_table(self, **kwargs):
             A new `astropy.table.Table` containing the data from the `~sunpy.timeseries.TimeSeries`.
             The table will include units where relevant.
         """
-        # TODO: Table.from_pandas(df) doesn't include the index column. Add request?
-        # Get data columns
-        table = Table.from_pandas(self._data)
-
-        # Get index column and add to table.
-        index_col = Column(self._data.index.values, name='date')
-        table.add_column(index_col, index=0)
-
-        # Add in units.
-        for key in self.units:
-            table[key].unit = self.units[key]
-
-        # Output the table
-        return table
+        # TODO: deprecate this
+        return Table(data=self._data, units=self.units)
```

Comment:
Can't we just return `.data` here? Timeseries is a subclass of Table?

Reply:
I think I tried that and some tests failed.

Reply:
The reason is that `self._data` doesn't have any units attached. I've deliberately done that to make this PR easier to review, but have another PR in the works to move the units on to self._data. At that point we can return ._data.